Housekeeping: remove unneeded legacy version checks & clean up code.

Moderocky opened the pull request:
## Description
<!--- Describe your changes here. --->
This is the batch merge request for #6641.

It removes checks for methods existing that are standard as of 1.13.2, since we no longer support versions below this.

### What can be removed?
Version checks for 1.13 and lower. Method & class existence checks that were standard as of **Spigot** for 1.13.2. Paper-only checks are still required for a few things that didn't arrive in Spigot until later (or at all).

See the linked issue #6641 for more details.

If you want to contribute to this project, open a pull request targeting `remove-legacy-code` and it will be included in this pull request.

### What next?

Another clean-up can be done before 2.10 for things under 1.17.
This will include a few bits and pieces that couldn't be fully cleaned up in this PR.

### This is a branch merge request!
Individual pull requests were made by contributors to its feature branch: this can be merged as-is to preserve the Git change history.

---
**Target Minecraft Versions:** any <!-- 'any' means all supported versions -->
**Requirements:** none <!-- Required plugins, server software... -->
**Related Issues:** #6641 <!-- Links to related issues -->

Moderocky committed:
Remove 1.13 method/version checks from conditions. (#6674)

Moderocky committed:
Remove 1.13 version checks from effects. (#6675)

* Remove deprecated visibility method use.

* Remove support checks from block change.

* Fix mistake in SendTitle (and remove 1.13 warnings).

* Remove check from make fly.

cheeezburga commented on src/main/java/ch/njol/skript/conditions/CondHasResourcePack.java:
```diff
@@ -37,8 +37,7 @@
 	static {
-		if (Skript.methodExists(Player.class, "hasResourcePack"))
-			register(CondHasResourcePack.class, PropertyType.HAVE, "[a] resource pack [(loaded|installed)]", "players");
+		register(CondHasResourcePack.class, PropertyType.HAVE, "[a] resource pack [(loaded|installed)]", "players");
```
Doesn't this still require the check? I can't see the Player#hasResourcePack() in Spigot 1.13.2.

Fusezion replied:
Based on the fact `@RequiredPlugins` state "Paper 1.9 or newer" it is possible. As such I did go and check latest spigot releases and there's no method for `hasResourcePack` so this check should still remain

sovdeeth replied:
yes this was only on the list for documentation update

cheeezburga commented on src/main/java/ch/njol/skript/conditions/CondIgnitionProcess.java:
```diff
@@ -42,12 +42,10 @@
-					"[creeper[s]] %livingentities% ((is|are)|1¦(isn't|is not|aren't|are not)) in the (ignition|explosion) process",
-					"creeper[s] %livingentities% ((is|are)|1¦(isn't|is not|aren't|are not)) ignited");
-		}
+		Skript.registerCondition(CondIgnitionProcess.class,
+			"[creeper[s]] %livingentities% ((is|are)|1¦(isn't|is not|aren't|are not)) going to explode",
+			"[creeper[s]] %livingentities% ((is|are)|1¦(isn't|is not|aren't|are not)) in the (ignition|explosion) process",
+			"creeper[s] %livingentities% ((is|are)|1¦(isn't|is not|aren't|are not)) ignited");
```
This as well.

cheeezburga committed:
Remove redundant exists checks in some expressions (#6691)

Removed some redundant existing checks and some other changes
- Happy to revert other changes (changes that aren't only the existing checks) if they're out of the scope of this PR (6676)

Moderocky committed:
Merge branch 'dev/feature' into remove-legacy-code

cheeezburga committed:
Removed some redundant exists checks and formatting stuff (#6696)

* Removed some redundant exists checks and made formatting more consistent in SimpleEvents

* Removed unused imports

* Put back the potiondata method handles

---

Co-authored-by: Moderocky <admin@moderocky.com>

cheeezburga committed:
Removed PassengerUtils class and replaced methods in referencing classes (#6693)

* Removed PassengerUtils class and replaced methods in referencing classes

* Apply suggestions from code review

* Formatting stuff

---------

Co-authored-by: Moderocky <admin@moderocky.com>

cheeezburga commented:
Can this PR just be used for the 2.10 stuff as well, seeing as it wasn't merged with 2.9?

cheeezburga committed:
Merge conflicts

cheeezburga committed:
Merge branch 'dev/feature' into remove-legacy-code
